ignite/metrics/gan/fid.py

```diff
@@ -0,0 +1,218 @@
+import warnings
+from distutils.version import LooseVersion
+from typing import Callable, Optional, Sequence, Union
+
+import numpy as np
+import torch
+
+from ignite.metrics.metric import Metric, reinit__is_reduced, sync_all_reduce
+
+__all__ = [
+    "FID",
+]
+
+
+def fid_score(
+    mu1: torch.Tensor, mu2: torch.Tensor, sigma1: torch.Tensor, sigma2: torch.Tensor, eps: float = 1e-6
+) -> Union[torch.Tensor, float]:
+
+    try:
+        import scipy
+    except ImportError:
+        raise RuntimeError("fid_score requires scipy to be installed.")
+
+    mu1, mu2 = mu1.cpu(), mu2.cpu()
+    sigma1, sigma2 = sigma1.cpu(), sigma2.cpu()
+
+    diff = mu1 - mu2
+
+    # Product might be almost singular
+    covmean, _ = scipy.linalg.sqrtm(sigma1.mm(sigma2), disp=False)
+    if not np.isfinite(covmean).all():
+        offset = np.eye(sigma1.shape[0]) * eps
+        covmean = scipy.linalg.sqrtm(((sigma1 * offset) * (sigma2 * offset)) / (eps * eps))
+
+    # Numerical error might give slight imaginary component
+    if np.iscomplexobj(covmean):
+        if not np.allclose(np.diagonal(covmean).imag, 0, atol=1e-3):
+            m = np.max(np.abs(covmean.imag))
+            raise ValueError("Imaginary component {}".format(m))
+        covmean = covmean.real
+
+    tr_covmean = np.trace(covmean)
+
+    return diff.dot(diff) + np.trace(sigma1) + np.trace(sigma2) - 2 * tr_covmean
+
+
+class InceptionExtractor:
+    def __init__(self) -> None:
+        try:
+            from torchvision import models
+        except ImportError:
+            raise RuntimeError("This module requires torchvision to be installed.")
+        self.model = models.inception_v3(pretrained=True)
+        self.model.fc = torch.nn.Identity()
+        self.model.eval()
+
+    @torch.no_grad()
+    def __call__(self, data: torch.Tensor) -> torch.Tensor:
+        if data.dim() != 4:
+            raise ValueError(f"Inputs should be a tensor of dim 4, got {data.dim()}")
+        if data.shape[1] != 3:
+            raise ValueError(f"Inputs should be a tensor with 3 channels, got {data.shape}")
+        return self.model(data)
+
+
+class FID(Metric):
+    r"""Calculates Frechet Inception Distance.
+
+    .. math::
+       \text{FID} = |\mu_{1} - \mu_{2}| + Trace(\sigma_{1} + \sigma_{2} - {2}\sqrt{\sigma_1*\sigma_2})
+
+    where :math:`mu1` and :math:`sigma1` refer to the mean and covariance of the train data and
+    :math:`mu2` and :math:`sigma2` refer to the mean and covariance of the test data.
```

Comment:
Please replace `Trace` with something from LaTeX, `\text{Tr}`, ":math:`mu1`" and ":math:`sigma1`"